binding/web/template/src/utils.ts

```diff
@@ -13,21 +13,21 @@
  * Convert a null terminated phrase stored inside an array buffer to a string
  *
  * @param arrayBuffer input array buffer
- * @param index the index at which the phrase is stored
+ * @param indexStart the index at which the phrase is stored
  * @return retrieved string
  */
 
- export function arrayBufferToStringAtIndex(
-  arrayBuffer: Uint8Array,
-  index: number,
-): string {
-  let stringBuffer = '';
-  let indexBuffer = index;
-  while (arrayBuffer[indexBuffer] !== 0) {
-    stringBuffer += String.fromCharCode(arrayBuffer[indexBuffer++]);
+  export function arrayBufferToStringAtIndex(
+    arrayBuffer: Uint8Array,
+    indexStart: number,
+  ): string {
+    let indexEnd = indexStart;
+    while (arrayBuffer[indexEnd] !== 0) {
+      indexEnd++;
+    }
+    let utf8decoder = new TextDecoder(); 
```

Comment:
I know the default is `utf-8` but let's be explicit here:

```suggestion
    let utf8decoder = new TextDecoder("utf-8"); 
```